diff --git a/dist/adapters/cache.js b/dist/adapters/cache.js
index d93d73f135a82e4ab89ca657bcab44de80bd8f1b..169c03498ff44978d1818fe35368684a23d355c6 100644
--- a/dist/adapters/cache.js
+++ b/dist/adapters/cache.js
@@ -116,7 +116,7 @@ export default class Cache {
                     const segmentData = new Map();
                     if (cacheData.segmentData) {
                         for (const [segmentPath, segmentContent] of Object.entries(cacheData.segmentData ?? {})) {
-                            segmentData.set(segmentPath, Buffer.from(segmentContent));
+                            segmentData.set(segmentPath, Buffer.from(segmentContent ?? ""));
                         }
                     }
                     return {
@@ -124,7 +124,7 @@ export default class Cache {
                         value: {
                             kind: "APP_PAGE",
                             html: cacheData.html,
-                            rscData: Buffer.from(cacheData.rsc),
+                            rscData: cacheData.rsc != null ? Buffer.from(cacheData.rsc) : undefined,
                             status: meta?.status,
                             headers: meta?.headers,
                             postponed: meta?.postponed,
@@ -232,7 +232,7 @@ export default class Cache {
                         await globalThis.incrementalCache.set(key, {
                             type: "app",
                             html,
-                            rsc: rscData.toString("utf8"),
+                            rsc: rscData != null ? rscData.toString("utf8") : undefined,
                             meta: {
                                 status,
                                 headers,
